src/app/components/service/service.component.ts

```diff
@@ -131,26 +130,19 @@ export class ServiceComponent implements OnInit, OnDestroy {
 
     this.subscription.add(
       combineLatest([
-        this.form.controls.status.valueChanges.pipe(startWith(undefined)),
+        this.form.controls.status.valueChanges.pipe(startWith([])),
         this.form.controls.search.valueChanges.pipe(
           startWith(''),
           map((v) => v.toLowerCase()),
         ),
-      ]).subscribe(([status, search]) => {
-        this.filteredTranslations$ = computed(() => {
-          let filtered = !status ? this.translations$() : this.filterMessages(this.translations$(), status.value)
-
-          this.animationState = !status ? 'in' : 'out'
-
-          this.filtered = status ? true : false
-
-          if (search.length > 0) {
-            filtered = this.filterMessages(filtered, undefined, search)
-          }
-
-          return filtered
-        })
-      }),
+      ]).subscribe(
+        ([status, search]) =>
+          (this.filteredTranslations$ = computed(() => {
+            this.animationState = status?.length === 0 ? 'in' : 'out'
+
+            return this.filterMessages(this.translations$(), status, search)
+          })),
```

Comment:
If `animationState` depends only on `status` changes, create a separate subscription that listens to `status` changes.

I used `status.length > 0 ? 'out' : 'in')` as it is shorter than `status.length === 0 ? 'in' : 'out'`.

```typescript
subscription.add(
  this.form.controls.status.valueChanges.pipe(startWith([])).subscribe(v => this.animationState = status.length > 0 ? 'out' : 'in')
)
```

```suggestion
          (this.filteredTranslations$ = computed(() => this.filterMessages(this.translations$(), status, search))),
```